logic/src/commonMain/kotlin/com/wire/kalium/logic/data/user/UserMapper.kt

```diff
@@ -82,8 +86,26 @@ interface UserMapper {
     ): UserEntity
 
     fun fromFailedUserToEntity(userId: NetworkQualifiedId): UserEntity
+    fun fromSelfUserToUserDetailsEntity(selfUser: SelfUser): UserDetailsEntity
+    fun fromSelfUserDtoToUserDetailsEntity(userDTO: SelfUserDTO): UserDetailsEntity
+    fun fromTeamMemberToDaoDetailModel(
+        teamId: TeamId,
+        nonQualifiedUserId: NonQualifiedUserId,
+        permissionCode: Int?,
+        userDomain: String
+    ): UserDetailsEntity
+
+    fun fromUserProfileDtoToUserDetailsEntity(
+        userProfile: UserProfileDTO,
+        connectionState: ConnectionEntity.State,
+        userTypeEntity: UserTypeEntity
+    ): UserDetailsEntity
+
+    fun fromFailedUserToDetailEntity(userId: NetworkQualifiedId): UserDetailsEntity
```

Comment:
suggestion: since it's `details` not `detail` I suggest to keep the name as it is in the function naming.